go-controller/pkg/util/util.go

```diff
@@ -362,7 +362,8 @@ func GetActiveNetworkForNamespace(namespace string, nadLister nadlister.NetworkA
 	for _, nad := range namespaceNADs {
 		nadInfo, err := ParseNADInfo(nad)
 		if err != nil {
-			return activeNetwork, err
```

Comment:
shouldn't we differentiate between a nonOVN managed NAD Parse error and other errors? if its other parse errors do we want to ignore them? (maybe we want to.. just checking if this was intentional)

Reply:
by that I mean "failed parsing NAD e2e-test-bond-vn7br/macvlannad: error parsing Network Attachment Definition e2e-test-bond-vn7br/macvlannad: net-attach-def not managed by OVN" we know we should continue on this error, but should we be continuing on all errors?

Reply:
We cannot really differenciate, the most we can do is do a warning log.

Reply:
> by that I mean "failed parsing NAD e2e-test-bond-vn7br/macvlannad: error parsing Network Attachment Definition e2e-test-bond-vn7br/macvlannad: net-attach-def not managed by OVN" we know we should continue on this error, but should we be continuing on all errors?

I have change the PR to continue on all the errors, since all of them affect this.

Reply:
I am not sure what you mean when you say "all of them affect's this"; so let's say I create a NAD with a bad pod subnet I'd rather have the pod creation fail over it than silently ignore it and go ahead with plumbing some secondary network... right?
so I was hoping for a well known type for that particular parse error

Reply:
Then you can create a non primary ovn NAD at a namespace with bad subnet and it will prevent any pod from scheduling, I don't know if this is what we really want.

I think what we really want is some kind of iterator that instead of doing full parse+validation just return ovn NADs with role=primary, then we do a full parse+validation and if it fails it make sense to prevent pod scheduling at that namespace.

Reply:
> ```
> Then you can create a non primary ovn NAD at a namespace with bad subnet and it will prevent any pod from scheduling, I don't know if this is what we really want.
> ```
> 
> IMHO that is how it should be. bad configs mean bad things.

Maybe, don't know if a failing scheduling pod not referencing that NAD is what the user wants, what is true though is that in a perfect world those errors apperas at NAD creation.

Reply:
From a UX perspective, this will be solved in the API level in the CRD - which we tell the users to use. 

This error should be indicated in the new CRD status.

Assuming we operate at NAD level, I think most we can do is indeed fail to start the pod, and throw an event.